### ISSUE TITLE
Issue with maxunits

### ISSUE THREAD

**Issue by Ealb** [elmokki/nationgen#64]
EA seed -151312420

Exception in thread "Thread-2" java.lang.NullPointerException
	at nationGen.rostergeneration.TroopGenerator.getNewTemplate(TroopGenerator.java:660)
	at nationGen.rostergeneration.TroopGenerator.generateUnit(TroopGenerator.java:294)
	at nationGen.rostergeneration.RosterGenerator.execute(RosterGenerator.java:152)
	at nationGen.nation.Nation.generate(Nation.java:197)
	at nationGen.nation.Nation.<init>(Nation.java:105)
	at nationGen.NationGen.generate(NationGen.java:209)
	at nationGen.NationGen.generate(NationGen.java:170)
	at nationGen.GUI.GUI$5.run(GUI.java:413)

I managed to trace through where this is happening, and confirm that it's maxunits failing to be properly dealt with in getNewTemplate; the check at line 637 finds that there are already 2 units when maxunits == 1, sets armor (formerly "barechested") to null, breaks w/o generating a unit, and proceeds to blow up at line 660 since u == null. It's not immediately obvious to me what the appropriate response here would be, though. The check against nulls at 654 can't be uncommented w/o merely changing where we blow up.

I tried uncommenting lines 654 and 655, and replacing line 294:

templates.add(getNewTemplate(race, role));

with:

Template temp = getNewTemplate(race, role);
if(temp != null) templates.add(temp);

but while this fixed the test case and weathered ~250 test nations, somewhere between the 250st and 300th it hung. For the moment I'm stepping back from this in hopes the answer is more obvious to you than me, but if not I'll dig into generateUnit to figure out where the infinite loop is cropping up, and how to avoid it outside of ignoring maxunit tags.

**Comment by elmokki** [comment]
I get Simbaea. No crash. Did you actually have my branch synced while you tested it because at least you haven't pulled it to your branch.

Regardless, I'll generate a few hundred nations to make sure.

**Comment by elmokki** [comment]
...yeah, something different popped up and this serious (EA -1816253951, I'm merging my branch in in a few): 

CRITICAL ERROR: No possible pose for Hoburg mages for tier 1.

Your hoburg themes are killing hoburgs as secondary race: They have no poses loaded without themes so they cannot be gotten as a mage of secondary race at all. Adding poses in themes isn't usable as liberally as it would seem.  

Now, there are two ways to approach this:
a) You move to chanceincs for poses. They're basechance 0 but themeincs raise it and for secondary races we'll add #secondaryracebasechance
b) Themes are race specific and optionally a secondary race theme set can be used

I'm gonna go for b) since that seems like a very sensible thing and it does make a lot of sense to not  , say, automatically always effectively make your secondary race cavemen advanced when your main race lizards are advanced.

**Comment by Ealb** [comment]
Hmmph. Well. I pulled everything before, but it doesn't look like I got everything synced to my desktop even though I was pretty sure I had. Bah, not sure how I managed that. I'm getting Simbaea now too. Well, at least this brought the issue with the hoburg themes to the fore. And yes, I'd agree that the ideal would be to go with the ... although it might be nice for themes to be able to set a chanceinc against the primary race's theme; e.g., Wolf Tribe barbarians are more likely to have Wolf Tribe vaetti allies than Spider Tribe... but cavemen aren't necessarily advanced just because the hoburgs who ride them are.
